### PR TITLE
Reorganize Canopy for export

### PR DESCRIPTION
**Pull request by dplumb94**
This PR reorganizes most of the Canopy code. The goal is to get Canopy to a state where it can be imported as a library to any React app to make it a Canopy application.

This PR implements a new component from CanopyJS, `CanopyProvider`, which provides the loading and config functionality to a canopy app. Most of the code for this comes directly from canopy/app, but it is reformatted to be a [React Context Provider](https://reactjs.org/docs/context.html).

### CODE REVIEW

File: canopy/canopyjs/src/index.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+  return userSaplingsResponse;
+};
+
+const fetchConfigSaplings = async saplingURL => {
+  const response = await get(`${saplingURL}/configSaplings`);
+  const configSaplingsResponse = response.json;
+  return configSaplingsResponse;
+};
+
+export function CanopyProvider({ saplingURL, splinterURL, children }) {
```
Comment (dnewh) [inline canopy/canopyjs/src/index.js]: Could this and its associated methods get moved into a canopy-provider.js file?

Comment (dplumb94) [reply canopy/canopyjs/src/index.js]: Is there a specific rationale for that? CanopyProvider and the useContext wrappers are the only things that canopyJS is going to export, so I didn't think there was much of a point to move them to a separate file.

File: canopy/canopyjs/README.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+- Exposes shared configuration to saplings and Canopy application components
+
+## Configuration
+
+CanopyJS makes use of two endpoints, `splinterURL` and `saplingURL`.
+
+### splinterURL
+
+`splinterURL` is the URL where the Splinter daemon is running. This URL will be
+used by Canopy and saplings to interact with Splinter via the Splinter Daemon's
```
Comment (chenette) [inline canopy/canopyjs/README.md]: daemon's

File: canopy/canopyjs/src/index.js (hunk 1 above)
Comment (dplumb94) [reply canopy/canopyjs/src/index.js]: After an in person discussion, we decided to resolve this in a future PR.

File: canopy/app/saplings/userSaplings [hunk 3]
```diff
@@ -0,0 +1,12 @@
+[
+  {
+    "displayName": "Vanilla",
+    "namespace": "vanilla",
+    "runtimeFiles": [
+      "0.0.0.0:8675/vanilla.js"
+    ],
+    "styleFiles": [],
+    "workerFiles": [],
+    "icon": "https://via.placeholder.com/48"
```
Comment (jsmitchell) [inline canopy/app/saplings/userSaplings]: is it possible to have this stuff self-hosted out of a local directory?

File: canopy/canopyjs/src/loadSaplings.js [hunk 4]
```diff
@@ -14,11 +14,10 @@
...
-import promiseLoader from 'promiseLoader';
-import styleLoader from 'styleLoader';
-import { getUserSaplings, getConfigSaplings } from './getSaplings.macro';
+import promiseLoader from './promiseLoader';
+import styleLoader from './styleLoader';
 
-async function loadCurrentSapling(userSaplingsResponse) {
+export async function mountCurrentSapling(userSaplingsResponse) {
   // Saplings will be guaranteed to to have a collision-free namespace that
   // coresponds with the place entry in the URL's path
```
Comment (jsmitchell) [inline canopy/canopyjs/src/loadSaplings.js]: I know it's not in the diff, but s/coresponds/corresponds/

File: canopy/app/saplings/userSaplings (hunk 3 above)
Comment (dplumb94) [reply canopy/app/saplings/userSaplings]: The icon specifically? In practice it would come from wherever the sapling is being hosted.

Comment (jsmitchell) [reply canopy/app/saplings/userSaplings]: Per in-person conversation this was converted from an existing json example. I'm ok deferring this for a future story which does a pass on self hosting as many assets as are reasonable (gameroom vue icons, this, etc.)

**Pull request merged**